app/scripts/services/images.js

```diff
@@ -2,6 +2,17 @@
 
 angular.module("openshiftConsole")
   .factory("ImagesService", function($filter, ApplicationGenerator, DataService) {
+
+    // maps an env object: { key: 'val', key2: 'val2'}
+    // to an array: [{},{},{}]
+    var makeEnvArray = function(input) {
+      return _.isArray(input) ?
+              input :
+              _.map(input, function(value, key){
+                return {name: key, value: value};
```

Comment:
This is just a switch for backwards compatibility.
- if it gets a hash, this is the "old way" that needs to transform into `[{name:'',value:''}, etc]`
-  If it gets an array, return the array (input arg).  We've already  

Off the top of my head I'm not sure why it received a hash of `{'<name>': '<value>'}` to begin with.